packages/babel-parser/src/parser/expression.js

```diff
@@ -2611,4 +2628,24 @@ export default class ExpressionParser extends LValParser {
 
     return ret;
   }
+
+  // https://github.com/tc39/proposal-js-module-blocks
+  parseModuleExpression(): N.ModuleExpression {
+    this.expectPlugin("jsModuleBlocks");
+    const node = this.startNode<N.ModuleExpression>();
+    node.body = [];
+    this.next(); // eat "module"
+    const oldLabels = this.state.labels;
+    this.state.labels = [];
+    const oldExportedIdentifiers = this.state.exportedIdentifiers;
+    this.state.exportedIdentifiers = [];
+    this.eat(tt.braceL);
+    this.scope.enter(SCOPE_OTHER);
```

Comment:
We should also enter a new production parameter scope with `[Await]` parameter when `topLevelAwait` is enabled, e.g.
```js
module { await 42 }
```

Reply:
189bc49 👍